packages/smooth_app/lib/main.dart

```diff
@@ -53,54 +53,44 @@ class _SmoothAppState extends State<SmoothApp> {
   late LocalDatabase _localDatabase;
   late ThemeProvider _themeProvider;
   bool systemDarkmodeOn = false;
-
-  Future<void> _init(BuildContext context) async {
-    _userPreferences = await UserPreferences.getUserPreferences();
-    _productPreferences = ProductPreferences(
-      ProductPreferencesSelection(
-        setImportance: (
-          String attributeId,
-          String importanceId,
-        ) async =>
-            _userPreferences.setImportance(attributeId, importanceId),
-        getImportance: (String attributeId) =>
-            _userPreferences.getImportance(attributeId),
-        notify: () => _productPreferences.notifyListeners(),
-      ),
-    );
-    try {
-      await _productPreferences.loadReferenceFromAssets(
-        DefaultAssetBundle.of(context),
-      );
-    } catch (e) {
-      // this is problematic - we should always be able to load the default
-      debugPrint('Could not load reference files: $e');
-      rethrow;
-    }
-    await _userPreferences.init(_productPreferences);
-    try {
-      _localDatabase = await LocalDatabase.getLocalDatabase();
-    } catch (e) {
-      // this is problematic - we should always be able to init the database
-      debugPrint('Cannot init database: $e');
-      rethrow;
-    }
-    _themeProvider = ThemeProvider(_userPreferences);
-  }
+  late Future<void> _initFuture;
```

Comment:
I would add a comment like `// Needed for hot reload; cf. https://github.com/openfoodfacts/smooth-app/issues/473`